File: programme/binary-search/binary-search.py
# Write a programme for binary search

def binarySearch(arr, target):
    low = 0
    high = len(arr) - 1
    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid - 1
    return -1


# get array from user
arr = input("Enter array  : ").split()
arr = [int(i) for i in arr]
target = int(input("Enter target : "))

if binarySearch(arr, target) == -1:
    print("\nOutput : Target not found")
else:
    print("\nOutput :", binarySearch(arr, target))

File: programme/binary-search/binary-search.java
// Write a programme for a binary search

public int binarySearch(int[] array, int target){
    int start = 0;
    int end = array.length - 1;
    while(start <= end){
        int mid = (start + end)/2;
        int midValue = array[mid];
        if(midValue < target){
            start = mid + 1;
        } else if(midValue > target){
            end = mid - 1;
        } else {
            return mid;
        }
    }
    return -start - 1;
}